src/Standard/TIP-3/1.md

```diff
@@ -0,0 +1,175 @@
+- This standard does not define the external methods to initiate transfer, mint or burn tokens.
+  Though it defines the methods, which MUST be called on a recipient token wallet or token root during these operations.
+- The rules for a token wallet ownership MUST be defined in a child standards.
+- A `-1` offset is added to some function IDs derivations, so the preimage of the hash cannot be known, further reducing the chances of a possible collisions.
+
+### Token root
+
+#### Name
+
+Returns the name of the token - e.g. "MyToken".
+
+```solidity
+function name() public view responsible returns (string);
+```
+
+#### Symbol
+
+Returns the symbol of the token. E.g. “HIX”.
+
+```solidity
+function symbol() public view responsible returns (string);
+```
+
+#### Decimals
+
+Returns the number of decimals the token uses - e.g. 8, means to divide the token amount by 100000000 to get its user representation.
+
+```solidity
+function decimals() public view responsible returns (uint8);
+```
+
+#### Total supply
+
+Returns the total token supply.
+
+```solidity
+function totalSupply() public view responsible returns (uint128);
+```
+
+#### Token wallet code
+
+Returns the token wallet code.
+
+```solidity
+function walletCode() public view responsible returns (TvmCell);
+```
+
+#### Accept tokens burn
+
+Does not have a standard signature, but has a standard function ID `0x192B51B1` obtained as `tvm.functionId('acceptBurn(uint128)') - 1`. The `uint128 _value` parameter MUST be first. The function name and the rest of the parameters are not fixed by this standard and can be reinvented for each substandard.
+
+Decreases token total supply by `_value`. The contract MUST check that the sender is a correct token wallet. Before sending this message, caller token wallet MUST decrease its own balance by `_value`. If the mint can't be accepted (e.g. invalid sender), this message MUST be bounced.
+
+Any function from the following snippet is a valid example:
+
+```solidity
+interface TIP3AcceptBurn {
+    function acceptBurn(uint128 _value) functionID(0x192B51B1) external;
+    function acceptBurn2(uint128 _value, uint256 _publicKey, address _owner) functionID(0x192B51B1) external;
+    function acceptBurn3(uint128 _value, TvmCell _meta) functionID(0x192B51B1) external;
+}
+```
+
+#### Standard interface detection
+
+```solidity
+interface TIP3TokenRoot {
+    /// @dev Accept burn function ID `0x192B51B1`
+    
+    function name() public view responsible returns (string);
+    function symbol() public view responsible returns (string);
+    function decimals() public view responsible returns (uint8);
+    function totalSupply() public view responsible returns (uint128);
+    function walletCode() public view responsible returns (TvmCell);
+}
+```
+
+The token root interface ID is `0x4371D8ED`.
+
+### Token wallet
+
+#### Root
+
+Returns the token root address.
+
+```solidity
+function root() public view responsible returns (address);
```

Comment:
I think we should give a name to return argument like `address addr`.

Reply:
I propose to keep it simple, the same as [EIP-20](https://eips.ethereum.org/EIPS/eip-20). The no-argument single-return getters don't have a name for return since the function name describes it itself. I mean, the return for `root()` is obviously a root address, but you can't even use the `root` name in return since it shadows a previous declaration.